### PR TITLE
Taint cookies, headers and path parameters

### PR DESCRIPTION
**Pull request by CarlesDD**
### What does this PR do?
Provides the feature to taint values coming from cookies, headers and path parameters.

### Motivation
Include more sources for tainting, improving custom code vulnerability detection

### CODE REVIEW

File: packages/dd-trace/src/appsec/iast/index.js [hunk 1]
```diff
@@ -48,6 +54,7 @@ function onIncomingHttpRequestStart (data) {
           const iastContext = iastContextFunctions.saveIastContext(store, topContext, { rootSpan, req: data.req })
           createTransaction(rootSpan.context().toSpanId(), iastContext)
           overheadController.initializeRequestContext(iastContext)
+          taintTrackingPlugin._headersTaintTrackingHandler(data.req.headers)
```
Comment (iunanua) [inline packages/dd-trace/src/appsec/iast/index.js:57]: rename `_headersTaintTrackingHandler` to be "public"?

File: packages/dd-trace/src/appsec/iast/taint-tracking/plugin.js [hunk 2]
```diff
@@ -16,18 +24,38 @@ class TaintTrackingPlugin extends Plugin {
...
+      ({ req }) => this._taintTrackingHandler(HTTP_REQUEST_PATH_PARAM, req, 'params')
+    )
   }
 
   _taintTrackingHandler (type, target, property) {
     const iastContext = getIastContext(storage.getStore())
+    if (!target) return
     if (!property) {
       target = taintObject(iastContext, target, type)
     } else {
```
Comment (iunanua) [inline packages/dd-trace/src/appsec/iast/taint-tracking/plugin.js:44]:
```suggestion
    } else if (target[property]) {
          target[property] = taintObject(iastContext, target[property], type)
```

**Review by iunanua — APPROVED** [review]

**Comment by iunanua** [comment]
ah, check this [comment](https://github.com/DataDog/dd-trace-js/pull/2618#issuecomment-1353810427 ) and this [commit](https://github.com/DataDog/dd-trace-js/pull/2618/commits/e38f8a08cf7658fb4fdb469afd6dc5b3248691d8) from the ldap plugin addition. 
I suppose you should do the same because you are adding a new instrumentation for cookie module.

File: .github/workflows/appsec.yml [hunk 3]
```diff
@@ -70,3 +70,19 @@ jobs:
       - uses: ./.github/actions/node/latest
       - run: yarn test:appsec:plugins:ci
       - uses: codecov/codecov-action@v2
+
+  sourcing:
+    runs-on: ubuntu-latest
+    env:
+      PLUGINS: express|cookies
```
Comment (iunanua) [inline .github/workflows/appsec.yml:77]: cookies or cookie?

**Comment by CarlesDD** [comment]
@iunanua I'm not sure about the changes regarding your last comment (include cookie in API docs and typings). 

Seems like these files are about plugins, so my question is why do we should include it in those file, since we haven't created any new plugin but just an instrumentation?

**Comment by iunanua** [comment]
I'm not sure either :S 
Ask Thomas, please.

**Pull request closed without merging**